zeek/bin/spicyz.cc

```diff
@@ -0,0 +1,226 @@
+                 "  -o | --output-to <path>         Path for saving output.\n"
+                 "  -v | --version                  Print version information.\n"
+                 "  -A | --abort-on-exceptions      When executing compiled code, abort() instead of throwing HILTI "
+                 "exceptions.\n"
+                 "  -B | --show-backtraces          Include backtraces when reporting unhandled exceptions.\n"
+                 "  -C | --dump-code                Dump all generated code to disk for debugging.\n"
+                 "  -D | --compiler-debug <streams> Activate compile-time debugging output for given debug streams "
+                 "(comma-separated; 'help' for list).\n"
+                 "  -L | --library-path <path>      Add path to list of directories to search when importing modules.\n"
+                 "  -O | --optimize                 Build optimized release version of generated code.\n"
+                 "  -R | --report-times             Report a break-down of compiler's execution time.\n"
+                 "  -T | --keep-tmps                Do not delete any temporary files created.\n"
+                 "  -X | --debug-addl <addl>        Implies -d and adds selected additional instrumentation "
+                 "(comma-separated; see 'help' for list).\n"
+                 "\n"
+                 "Inputs can be *.spicy, *.evt, *.hlt, .cc/.cxx\n"
+                 "\n";
+}
+
+using hilti::Nothing;
+
+static hilti::Result<Nothing> parseOptions(int argc, char** argv, hilti::driver::Options* driver_options,
+                                           hilti::Options* compiler_options) {
+    while ( true ) {
+        int c = getopt_long(argc, argv, "ABc:CdX:D:L:o:ORTvh", long_driver_options, nullptr);
+
+        if ( c < 0 )
+            break;
+
+        switch ( c ) {
+            case 'A': driver_options->abort_on_exceptions = true; break;
+
+            case 'B': driver_options->show_backtraces = true; break;
+
+            case 'c':
+                driver_options->output_cxx = true;
+                driver_options->output_cxx_prefix = std::string(optarg);
+                break;
+
+            case 'C': {
+                driver_options->dump_code = true;
+                break;
+            }
+
+            case 'd': {
+                compiler_options->debug = true;
+                break;
+            }
+
+            case 'X': {
+                auto arg = std::string(optarg);
+
+                if ( arg == "help" ) {
+                    std::cerr << "Additional debug instrumentation:\n";
+                    std::cerr << "   flow:     log function calls to debug stream \"hilti-flow\"\n";
+                    std::cerr << "   location: track current source code location for error reporting\n";
+                    std::cerr << "   trace:    log statements to debug stream \"hilti-trace\"\n";
+                    std::cerr << "\n";
+                    exit(0);
```

Comment:
This is inconsistent with how we e.g., handle `'v'` or `'h'`. I'd say prefer not `exit`'ing from here, but on the top-level.
```suggestion
                    return Nothing();
```